design/login.md

```diff
@@ -0,0 +1,102 @@
+# login
+
+The login package handles DC/OS login related operations.
+
+## Goals
+
+The goals of the login package are to :
+
+- Detect available login providers for a DC/OS cluster.
+- Log in to a DC/OS cluster using a given login provider.
+
+## Login providers detection
+
+To detect available login providers for a given DC/OS cluster, the CLI issues a GET request to the
+`/acs/api/v1/auth/providers` endpoint. This endpoint returns a list of providers in JSON. Each
+top-level key refers to the login provider ID, while its value is the login provider definition:
+
+``` json
+{
+  "dcos-users": {
+    "authentication-type": "dcos-uid-password",
+    "client-method": "dcos-usercredential-post-receive-authtoken",
+    "config": {
+      "start_flow_url": "/acs/api/v1/auth/login"
+    },
+    "description": "Default DC/OS authenticator"
+  }
+}
+```
+
+The above definition is a default DC/OS login provider, its ID is **dcos-users**.
+The ID has no special semantic apart from uniquely identifying a login provider.
+
+When the `/acs/api/v1/auth/providers` endpoint is not present, the available login provider is
```

Comment:
Could you add for which DC/OS versions will this endpoint not be present?

Reply:
I've left it out on purpose as this is not relevant to the design of this feature. If we were to say for eg. `this is not available in DC/OS open source`, that document will become inaccurate at some point.